Assets/Scripts/UIManager.cs

```diff
@@ -69,4 +86,46 @@ public void LevelText(int currentLevel)
     {
         _levelText.text = "Level: " + currentLevel.ToString();
     }
+
+    public void OptionsMenuButton()
+    {
+        if (_isOptionsMenuOpen == false)
+        {
+            OpenOptionsMenu();
+        }
+        else if (_isOptionsMenuOpen == true)
+        {
+            CloseOptionsMenu();
+        }
+    }
+
+    private void OpenOptionsMenu()
```

Comment:
you can actually combine this method and the one below to be

```
    private void ShowOptionsMenu(bool active)
    {
        _restartButton.gameObject.SetActive(active);
        _premadeLevelButton.gameObject.SetActive(active);
        _optionsMenuBackground.gameObject.SetActive(active);
        _isOptionsMenuOpen = active;
        Time.timeScale = active ? 0 : 1;
    }
```

and then in your `OptionsMenuButton()` method above, you could do

```
ShowOptionsMenu(_isOptionsMenuOpen);
```

Reply:
even better, in the unity editor where you set up the button OnClick event, you can actually have it pass in a boolean to you so that you don't even need to maintain the `_isOptionsMenuOpen` bool yourself.